### PR TITLE
RCAL-169: Remove datamodels

### LINKED ISSUE #235: Remove stdatamodels as dependency
_Issue [RCAL-169](https://jira.stsci.edu/browse/RCAL-169) was created on JIRA by [James Davies](https://jira.stsci.edu/secure/ViewProfile.jspa?name=jdavies):_

1. Remove all stdatamodels imports in romancal and remove it from setup.cfg as a install dependency.
2. File issue in stdatamodels to remove it from the downstream tests on stdatamodels CI.
3. Update the release procedures for stdatamodels to no longer run romancal tests on tagging and release.

### PR DESCRIPTION
**Pull request by nden**
Closes #235

Resolves RCAL-169

**Description**

Roman datamodels are in the `roman_datamodels`  package. Removing the obsolete `romancal/datamodels` directory

This also removes direct dependency on stadatamodels. Stpipe still depends on stdatamodels for the s3_utils module
and so indirectly pull stdatamodels into a romancal environment.

Checklist

- [ ] Change log

- [ ] Milestone

- [ ] Label(s)

### CODE REVIEW

File: docs/roman/references_general/flat_reffile.inc [hunk 1]
```diff
@@ -4,7 +4,7 @@ FLAT Reference File
 -------------------
 
 :REFTYPE: FLAT
-:Data model: `~romancal.datamodels.FlatModel`
+:Data model: `roman_datamodels.FlatModel`
```
Comment (eslavich) [inline docs/roman/references_general/flat_reffile.inc:7]: This seems to be named [FlatRefModel](https://github.com/spacetelescope/roman_datamodels/blob/main/src/roman_datamodels/datamodels.py#L282)

**Review by eslavich — COMMENTED** [review]
Just the one thing about the name of the flat model.  Nice to say goodbye to all of this code!

File: docs/roman/references_general/flat_reffile.inc (hunk 1 above)
Comment (nden) [reply docs/roman/references_general/flat_reffile.inc]: fixed

**Review by eslavich — APPROVED** [review]
LGTM!

**Pull request merged**